### DISCUSSION TITLE
Ability to create transactions on-demand

### DISCUSSION THREAD

**Discussion by engram-design** [craftcms/commerce#2351] [Ideas]
We have a scenario where our client wants to create transactions on-demand. In this instance, its where a refund has been processed offline. Their previous CMS in fact had a status called "Capture Offline" that worked well for this.

Totally understand this can work for the Manual or Dummy payment gateway, but it'd be great if this could be used for other gateways.

For example, a customer pays with PayPal, but after 90 days you're unable to refund via the API. As such, you can't use the Refund facility in the Transactions tab. The shop owner needs to do this transaction manually through PayPal, but it should also be tracked in Commerce under the order.

Another scenario could be where you might receive a cash payment (partial or otherwise) and you want to record that against an order.

Essentially - it'd be nice to have a "New Transaction" button to be able to add these on-demand. I suppose for Stripe and other payment gateways, being able to arbitrarily capture payments would be handy as well.

**Comment by gojira32** [comment]
@lukeholder Any movement on this?

**Comment by lukeholder** [comment]
Nothing in the immediate term. We would need to introduce a new concept of a  gateway-specific 'manual' transaction. Will put it on the 3.x list.

**Comment by mylesthedev** [comment]
👀

**Comment by engram-design** [comment]
It'd also be swell to be able to delete transactions
